lib/play.rb

```diff
@@ -7,7 +7,7 @@
 class Play
   attr_reader :board, :cpu, :player, :game_over
 
-  def initialize(board, cpu, player, game_over, output: output)
+  def initialize(board, cpu, player, game_over, output: ui)
```

Comment:
Where is `ui` coming from? That's the default variable correct?

Reply:
Yes, it was causing a circular reference because it had the same name so I changed it to ui.

Reply:
Ah I understand what's going on now. 

The syntax you're using says "if `output` isn't specified use `ui` as the default". Only `ui` is undefined, so that would crash. This is only working because every time you use `Play` you also specify output. 

What you should do to get the interface you want is just have `def initialize(board, cpu, player, game_over, output:)` with the colon after output and nothing after that. That way you can still use keyword arguments for clarity, but there's no default value.

Reply:
This is a ruby thing:

function(output) -> You must specify a value, no keyword args